packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -24,6 +24,7 @@ import 'text_theme.dart';
 import 'theme.dart';
 
 const double _kLeadingWidth = kToolbarHeight; // So the leading button is square.
+const double _kMaxTitleTextScaleFactor = 1.34;
```

Comment:
This would be a good place for a link to the place in the MD spec where this limit is given.

Reply:
Added TODO for this, to add the link when the text scaling is added to the spec. https://github.com/flutter/flutter/issues/58769